### ISSUE TITLE
[prefer-optional-chain] Only checks the first occurrence in conditional

### ISSUE THREAD

**Issue by garyking** [typescript-eslint/typescript-eslint#1461]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/prefer-optional-chain": "warn"
  }
}
```

```TS
foo && foo.a && bar && bar.a
```

**Expected Result**

```ts
foo?.a && bar?.a
```

**Actual Result**

```ts
foo?.a && bar && bar.a
```

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.16.0` |
| `@typescript-eslint/parser`        | `2.16.0` |
| `TypeScript`                       | `3.7.4` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `13.5.0` |
| `npm`                              | `6.13.4` |

**Comment by bradzacher** [comment]
Related to #1438 - the selectors don't match everything. They're too relaxed atm.

**Comment by Josh-Cena** [comment]
Fixed by https://github.com/typescript-eslint/typescript-eslint/pull/6397
